base_revision/models/base_revision.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2013 Agile Business Group sagl (<http://www.agilebg.com>)
+# Copyright 2016 Serpent Consulting Services Pvt. Ltd.
+# Copyright 2018 Dreambits Technologies Pvt. Ltd. (<http://dreambits.in>)
+# Copyright 2020 Ecosoft Co., Ltd. (<http://ecosoft.co.th>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+
+
+class BaseRevision(models.AbstractModel):
+    _name = "base.revision"
+    _description = "Document Revision (abstract)"
+
+    @api.depends("old_revision_ids")
+    def _compute_has_old_revisions(self):
+        for rec in self:
+            rec.has_old_revisions = (
+                True if rec.with_context(active_test=False).old_revision_ids else False
+            )
```

Comment:
(Non-blocking comment) Could you reorganize the code? The correct order should follow the guidelines:
https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#37models

![image](https://github.com/user-attachments/assets/30beae41-5bf6-45e2-957c-528a88aa59b2)